File: 5th Semester/mobile/swift/Model/Reservation.swift
//
//  Reservation.swift
//  Rental
//
//  Created by Mircea Maierean on 05.11.2024.
//

import Foundation
import SwiftData

class Reservation {
    var id: UUID
    var carId: Int
    var userId: UUID
    var totalPrice: Int
    var startDate: Date
    var endDate: Date
    
    init(
        id: UUID = UUID(),
        carId: Int,
        userId: UUID,
        totalPrice: Int = 0,
        startDate: Date = Date(),
        endDate: Date = Date()
    ) {
        self.id = id
        self.carId = carId
        self.userId = userId
        self.totalPrice = totalPrice
        self.startDate = startDate
        self.endDate = endDate
    }
}

File: 5th Semester/mobile/flutter/rental/lib/models/reservation.dart
import 'package:uuid/uuid.dart';

class Reservation {
  final String id;
  final int carId;
  final String userId;
  final int totalPrice;
  final DateTime startDate;
  final DateTime endDate;

  Reservation({
    String? id,
    required this.carId,
    required this.userId,
    this.totalPrice = 0,
    DateTime? startDate,
    DateTime? endDate,
  })  : id = id ?? Uuid().v4(),
        startDate = startDate ?? DateTime.now(),
        endDate = endDate ?? DateTime.now();
}
